### PR TITLE
Fix #215, shorter build times on Travis

### LINKED ISSUE #215: fix long build/test times in Travis
* Travis jobs testing PRs (+ `master` on merges) now take a very long time, and sometimes fail because of crossing the 1h time limit
  * see for example https://travis-ci.org/hpcugent/vsc_user_docs/builds/421043323
  * `./install-tl -profile $TRAVIS_BUILD_DIR/texlive.profile` takes over 2600s...
* options to try:
  * use Travis cache rather than reinstalling every time (see https://docs.travis-ci.com/user/caching/)
  * use more recent Ubuntu image in Travis + install straight from Ubuntu repos

### PR DESCRIPTION
**Pull request by redfast00**

### CODE REVIEW

**Comment by boegel** [comment]
Current Travis job: https://travis-ci.org/hpcugent/vsc_user_docs/builds/421089746

@redfast00 When that Travis jobs completes, please try closing & re-opening this PR. I'm not sure that would already pick up on the cache though, it may need to be merged to `master` first...

In any case, we can certainly merge this and see how it works out, and revert if needed.

This will only be useful if `./install-tl` bothers to check what is already installed in `$HOME/.texlive`. If it doesn't and just re-installs blindly, there will be no benefit, and we should look into the other option.

**Comment by boegel** [comment]
@redfast00 Travis job failed:

```
$ pdflatex --version ; latexmk --version
The program 'pdflatex' is currently not installed. To run 'pdflatex' please ask your administrator to install the package 'texlive-latex-base'
The program 'latexmk' is currently not installed. To run 'latexmk' please ask your administrator to install the package 'latexmk'
The command "pdflatex --version ; latexmk --version" failed and exited with 127 during .
```

File: travis-install.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    cd
+}
+
+# See if there is a cached version of TL available
+export PATH=$CACHEDIR/.texlive/bin/x86_64-linux:$PATH
+if ! command -v pdflatex > /dev/null; then
+    echo "First install"
+    install_texlive
+else
+    if ! (( $RANDOM % 100 )); then
```
Comment (boegel) [inline travis-install.sh]: @redfast00 Maybe add a comment to clarify this, i.e. "`Force a cache update once every 100 builds (on average)`"

File: travis-install.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sed -i "s:MYPREFIX:$CACHEDIR:g" $TRAVIS_BUILD_DIR/texlive.profile
+
+    # Install a minimal system
+    ./install-tl -profile $TRAVIS_BUILD_DIR/texlive.profile
+    cd
+}
+
+# See if there is a cached version of TL available
+export PATH=$CACHEDIR/.texlive/bin/x86_64-linux:$PATH
+if ! command -v pdflatex > /dev/null; then
```
[the remark this answers is not in the document]
Comment (redfast00) [reply travis-install.sh]: It doesn't really make much of a difference in our usecase: https://stackoverflow.com/a/37056347/5431090

File: travis-install.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+set -e -u
+
+CACHEDIR="$HOME/cached"
```
[the remark this answers is not in the document]
Comment (redfast00) [reply travis-install.sh]: Good catch, will do

File: travis-install.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+install_texlive()
+{
+    cd /tmp
```
[the remark this answers is not in the document]
Comment (boegel) [reply travis-install.sh]: Yeah, good point. Also, the script itself probably isn't very big, so it wouldn't make much of a difference to cache it, I guess...

File: travis-install.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e -u
+
+travis_retry() {
```
[the remark this answers is not in the document]
Comment (redfast00) [reply travis-install.sh]: Do you want me to remove this?

[the remark this answers is not in the document]
Comment (boegel) [reply travis-install.sh]: Yes, I think copy-pasting it is a bad idea, then I'd rather not use it...

Or, we could not use a subshell for the script?

[the remark this answers is not in the document]
Comment (redfast00) [reply travis-install.sh]: https://twitter.com/plexus/status/499194992632811520 I'm afraid we cannot. Guess I'll just remove it.

File: travis-install.sh [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+set -e -u
+
+CACHEDIR="$HOME/cached"
+
+install_texlive()
+{
+    cd /tmp
+    # Obtain TeX Live
+    wget http://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz
```
[the remark this answers is not in the document]
Comment (boegel) [reply travis-install.sh]: (nevermind)

File: travis-install.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (boegel) [reply travis-install.sh]: TIL, `command -v` is indeed better since it avoids depending on `which`, I should start using that more often...

**Pull request merged**